### DISCUSSION TITLE
Disable global styles or specify root component

### DISCUSSION THREAD

**Discussion by cem2ran** [chakra-ui/chakra-ui#3960] [Q&A]
Hi all, really enjoying using Chakra, but ran into the issue of Chakra styles escaping the scope of the component I render my react component to. I'm using Chakra to build a chrome extension that injects a component to a div tag in the HTML body of a website, but I'm seeing Chakra styles being added to the whole website (html, body, text, etc...).

Is there a way to disable this behavior or specify the root of my Chakra application?

**Comment by ianrohde** [comment]
Similar issue. Transitioning an old codebase incrementally to use Chakra components exposes the same global element-level style collisions:

```html
<head>
  ...
  <style data-emotion="css-global">h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit;}</style>
  ...
</head>
```

I attempted to resolve by [attaching the CSS variables](https://chakra-ui.com/docs/features/css-variables#attaching-the-css-variables) to my outer Chakra elements, but was unsuccessful. @cem2ran, you might have better luck with that in your situation.

This seems linked to the Chakra CSS reset since it targets elements directly. Setting [`resetCSS={false}`](https://chakra-ui.com/docs/getting-started#chakraprovider-props) solves the global issue, but blocks Chakra's full component styling.

**Reply by ianrohde** [reply]
`cssVarsRoot` related discussion: https://github.com/chakra-ui/chakra-ui/discussions/4022
